cli/inputhandler/jsonfile.cpp

```diff
@@ -0,0 +1,43 @@
+#include "jsonfile.h"
+#include "utils/errors.h"
+
+#include <iostream>
+#include <fstream>
+
+namespace ih
+{
+JsonFile::JsonFile(std::string const &filePath) :
+        IFile(filePath)
+{
+    try
+    {
+        JsonFile::checkFile();
+    }
+    catch (std::exception const &error)
+    {
+        throw;
+    }
+}
+
+bool JsonFile::checkFile() const
+{
+    if (!IFile::fileExists()) throw std::invalid_argument(ERROR_MSG_FILE_DOES_NOT_EXIST);
+    if (!IFile::isFileExtension("json")) throw std::invalid_argument(ERROR_MSG_FILE_EXTENSION_INVALID);
+
+    return true;
+}
+
+void JsonFile::writeDataToFile(std::string const &data)
+{
+    std::ofstream myFile(IFile::getFilePath());
+
+    if (checkFile())
```

Comment:
I see we check the file again - also for its existence. Not sure if we should allow it to be missing, instead, and simply create it (instead of expecting an empty file). Oh, but then there would be changes in the constructor, since the constructor always checks for the existence of the file. Perhaps we can add 2 factory methods (`newInputFile()` and `newOutputFile` - similar to opening a file with a specific `FileMode`, after all) instead of the unique constructor and check for existence in the first one and for non-existence in the second one.

Can stay as it is though for the moment, we can think about this later.

Reply:
Good point. We should take care of it later.